File: app/src/main/java/com/example/neon/deardiary/DAO/Diary.java
package com.example.neon.deardiary.dao;

import org.greenrobot.greendao.annotation.Entity;
import org.greenrobot.greendao.annotation.Generated;
import org.greenrobot.greendao.annotation.Id;

import java.io.Serializable;
import java.util.Calendar;

/**
 * 日记实体对象
 *
 * Created by Neon on 2016/11/30.
 */

@Entity
public class Diary implements Serializable {
    private static final long serialVersionUID = 0L;

    //字段
    @Id
    private Long id;//数据库id
    private int year;//年
    private int month;//月
    private int dayOfMonth;//日
    private int dayOfWeek;//星期
    private int hour;//当天最后一次编辑的小时
    private int minute;//当天最后一次编辑的分钟
    private String title;//日记标题
    private String content;//日记内容


    @Generated(hash = 455145805)
    public Diary(Long id, int year, int month, int dayOfMonth, int dayOfWeek, int hour,
                 int minute, String title, String content) {
        this.id = id;
        this.year = year;
        this.month = month;
        this.dayOfMonth = dayOfMonth;
        this.dayOfWeek = dayOfWeek;
        this.hour = hour;
        this.minute = minute;
        this.title = title;
        this.content = content;
    }


    @Generated(hash = 112123061)
    public Diary() {
    }

    public Diary(Calendar c) {
        this.year = c.get(Calendar.YEAR);
        this.month = c.get(Calendar.MONTH) + 1;
        this.dayOfMonth = c.get(Calendar.DAY_OF_MONTH);
        this.dayOfWeek = c.get(Calendar.DAY_OF_WEEK);
        this.content = "";
        this.title = "";
        this.hour = 0;
        this.minute = 0;
    }


    //getters and setters
    public int getYear() {
        return year;
    }

    public void setYear(int year) {
        this.year = year;
    }

    public int getMonth() {
        return month;
    }

    public void setMonth(int month) {
        this.month = month;
    }

    public int getDayOfMonth() {
        return dayOfMonth;
    }

    public void setDayOfMonth(int dayOfMonth) {
        this.dayOfMonth = dayOfMonth;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }


    public Long getId() {
        return this.id;
    }


    public void setId(Long id) {
        this.id = id;
    }


    public int getDayOfWeek() {
        return this.dayOfWeek;
    }


    public void setDayOfWeek(int dayOfWeek) {
        this.dayOfWeek = dayOfWeek;
    }


    public int getHour() {
        return this.hour;
    }


    public void setHour(int hour) {
        this.hour = hour;
    }


    public int getMinute() {
        return this.minute;
    }


    public void setMinute(int minute) {
        this.minute = minute;
    }


    public String getTitle() {
        return this.title;
    }


    public void setTitle(String title) {
        this.title = title;
    }
}

File: app/src/main/java/com/wenhaiz/deardiary/data/Diary.kt
package com.wenhaiz.deardiary.data

import io.objectbox.annotation.Entity
import io.objectbox.annotation.Id
import java.io.Serializable
import java.util.Calendar

@Entity
class Diary : Serializable {
    @Id
    var id: Long = 0//数据库id
    var year: Int = 0//年
    var month: Int = 0//月
    var dayOfMonth: Int = 0//日
    var dayOfWeek: Int = 0//星期
    var hour: Int = 0//当天最后一次编辑的小时
    var minute: Int = 0//当天最后一次编辑的分钟
    var title: String = ""//日记标题
    var content: String = ""//日记内容

    constructor(id: Long, year: Int, month: Int, dayOfMonth: Int, dayOfWeek: Int, hour: Int,
                minute: Int, title: String, content: String) {
        this.id = id
        this.year = year
        this.month = month
        this.dayOfMonth = dayOfMonth
        this.dayOfWeek = dayOfWeek
        this.hour = hour
        this.minute = minute
        this.title = title
        this.content = content
    }

    constructor() {
        this.content = ""
        this.title = ""
        this.year = 0
        this.minute = 0
        this.hour = 0
        this.dayOfMonth = 0
        this.dayOfWeek = 0
        this.month = 0
    }

    constructor(c: Calendar) : this() {
        this.year = c.get(Calendar.YEAR)
        this.month = c.get(Calendar.MONTH) + 1
        this.dayOfMonth = c.get(Calendar.DAY_OF_MONTH)
        this.dayOfWeek = c.get(Calendar.DAY_OF_WEEK)
    }

    override fun equals(other: Any?): Boolean {
        if (other == null) {
            return false
        }
        return if (other is Diary) {
            this.year == other.year && this.month == other.month && this.dayOfMonth == other.dayOfMonth
        } else {
            false
        }
    }

    override fun hashCode(): Int {
        var result = id.hashCode()
        result = 31 * result + year
        result = 31 * result + month
        result = 31 * result + dayOfMonth
        result = 31 * result + dayOfWeek
        result = 31 * result + hour
        result = 31 * result + minute
        result = 31 * result + title.hashCode()
        result = 31 * result + content.hashCode()
        return result
    }

    companion object {
        private const val serialVersionUID = 1L

        fun emptyDiary(c: Calendar): Diary {
            return Diary(c)
        }
    }
}
